test_vsids failing: assert 2.0 > 2.0

Encountered on #6191. See [here](https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_win64_pytest/1604/console) for the console output.

```
[gw7] [ 99%] PASSED src/tribler-core/tribler_core/modules/remote_query_community/tests/test_eva_protocol.py::TestEVA::test_multiply_duplex 
src/tribler-core/tribler_core/modules/remote_query_community/tests/test_remote_query_community.py::TestRemoteQueryCommunity::test_process_rpc_query_match_many 
[gw7] [100%] PASSED src/tribler-core/tribler_core/modules/remote_query_community/tests/test_remote_query_community.py::TestRemoteQueryCommunity::test_process_rpc_query_match_many 

================================== FAILURES ===================================
_________________________________ test_vsids __________________________________
[gw8] win32 -- Python 3.8.4 c:\users\tribler\appdata\local\programs\python\python38\python.exe

metadata_store = <tribler_core.modules.metadata_store.store.MetadataStore object at 0x00000000295C6130>

    @db_session
    def test_vsids(metadata_store):
        """
        Test VSIDS-based channel popularity system.
        """
        peer_key = default_eccrypto.generate_key("curve25519")
        assert metadata_store.Vsids[0].bump_amount == 1.0
    
        channel = metadata_store.ChannelMetadata.create_channel('test', 'test')
        metadata_store.vote_bump(channel.public_key, channel.id_, peer_key.pub().key_to_bin()[10:])
        sleep(0.1)  # Necessary mostly on Windows, because of the lower timer resolution
        metadata_store.vote_bump(channel.public_key, channel.id_, peer_key.pub().key_to_bin()[10:])
        assert channel.votes > 0.0
        assert metadata_store.Vsids[0].bump_amount > 1.0
    
        # Make sure normalization for display purposes work
        assert channel.to_simple_dict()["votes"] == 1.0
    
        # Make sure the rescale works for the channels
        metadata_store.Vsids[0].normalize()
        assert metadata_store.Vsids[0].bump_amount == 1.0
        assert channel.votes == 1.0
    
        # Ensure that vote by another person counts
        peer_key = default_eccrypto.generate_key("curve25519")
        metadata_store.vote_bump(channel.public_key, channel.id_, peer_key.pub().key_to_bin()[10:])
        assert channel.votes == 2.0
    
        sleep(0.1)  # Necessary mostly on Windows, because of the lower timer resolution
        # Ensure that a repeated vote supersedes the first vote but does not count as a new one
        metadata_store.vote_bump(channel.public_key, channel.id_, peer_key.pub().key_to_bin()[10:])
>       assert channel.votes > 2.0
E       assert 2.0 > 2.0
E        +  where 2.0 = ChannelMetadata[1].votes

src\tribler-core\tribler_core\modules\metadata_store\tests\test_channel_metadata.py:356: AssertionError
```

I noticed that there is already something that looks like a workaround on Windows - maybe revise it?

--- a/src/requirements-test.txt
+++ b/src/requirements-test.txt
@@ -8,5 +8,7 @@ pytest-mock
 pytest-randomly                        
 pytest-timeout                         
 pytest-xdist
+pytest-freezegun
+freezegun
 
 asynctest

--- a/src/tribler-core/tribler_core/modules/metadata_store/orm_bindings/vsids.py
+++ b/src/tribler-core/tribler_core/modules/metadata_store/orm_bindings/vsids.py
@@ -70,23 +70,34 @@ def define_binding(db):
 
         @db_session
         def bump_channel(self, channel, vote):
+            now = datetime.utcnow()
+
             # Subtract the last vote by the same peer from the total vote amount for this channel.
             # This effectively puts a cap of 1.0 vote from a peer on a channel
             channel.votes -= vote.last_amount
             self.total_activity -= vote.last_amount
 
-            vote.last_amount = self.bump_amount
-            channel.votes += self.bump_amount
-            self.total_activity += self.bump_amount
+            # Next, increase the bump amount based on the time passed since the last bump
+            # (Increasing the bump amount is the equivalent of decaying the values of votes,
+            # cast for all other channels)
+            self.bump_amount *= math.exp((now - self.last_bump).total_seconds() / self.exp_period)
+            self.last_bump = now
 
+            # To cap the future votes from this peer, note the last bump vote
+            # amount added to this channel by the voting peer.
+            vote.last_amount = self.bump_amount
+
+            # Add the vote to the accumulated sum of all votes for the channel
+            channel.votes += self.bump_amount
+
+            # Keep track of total activity and max vote amount to assist with votes scaling/normalization
+            self.total_activity += self.bump_amount
             if channel.votes > self.max_val:
                 self.max_val = channel.votes
             db.ChannelMetadata.votes_scaling = self.max_val
 
-            bump_moment = datetime.utcnow()
-            self.bump_amount *= math.exp((bump_moment - self.last_bump).total_seconds() / self.exp_period)
-            self.last_bump = bump_moment
-
+            # Renormalize all votes in the database if current bump amount is nearing the float
+            # precision threshold
             if self.bump_amount > self.rescale_threshold:
                 self.rescale(self.bump_amount)
 

--- a/src/tribler-core/tribler_core/modules/metadata_store/tests/test_channel_metadata.py
+++ b/src/tribler-core/tribler_core/modules/metadata_store/tests/test_channel_metadata.py
@@ -3,7 +3,6 @@ from binascii import unhexlify
 from datetime import datetime
 from itertools import combinations
 from pathlib import Path
-from time import sleep
 from unittest.mock import Mock, patch
 
 from ipv8.keyvault.crypto import default_eccrypto
@@ -16,7 +15,6 @@ import pytest
 
 from tribler_common.simpledefs import CHANNEL_STATE
 
-from tribler_core.exceptions import DuplicateTorrentFileError
 from tribler_core.modules.libtorrent.torrentdef import TorrentDef
 from tribler_core.modules.metadata_store.orm_bindings.channel_metadata import (
     CHANNEL_DIR_NAME_LENGTH,
@@ -321,8 +319,9 @@ def test_correct_commit_of_delete_entries(metadata_store):
         channel.commit_channel_torrent()
 
 
+@pytest.mark.freeze_time('2021-09-24')
 @db_session
-def test_vsids(metadata_store):
+def test_vsids(metadata_store, freezer):
     """
     Test VSIDS-based channel popularity system.
     """
@@ -331,7 +330,7 @@ def test_vsids(metadata_store):
 
     channel = metadata_store.ChannelMetadata.create_channel('test', 'test')
     metadata_store.vote_bump(channel.public_key, channel.id_, peer_key.pub().key_to_bin()[10:])
-    sleep(0.1)  # Necessary mostly on Windows, because of the lower timer resolution
+    freezer.move_to('2021-09-25')
     metadata_store.vote_bump(channel.public_key, channel.id_, peer_key.pub().key_to_bin()[10:])
     assert channel.votes > 0.0
     assert metadata_store.Vsids[0].bump_amount > 1.0
@@ -349,11 +348,10 @@ def test_vsids(metadata_store):
     metadata_store.vote_bump(channel.public_key, channel.id_, peer_key.pub().key_to_bin()[10:])
     assert channel.votes == 2.0
 
-    sleep(0.1)  # Necessary mostly on Windows, because of the lower timer resolution
+    freezer.move_to('2021-09-26')
     # Ensure that a repeated vote supersedes the first vote but does not count as a new one
     metadata_store.vote_bump(channel.public_key, channel.id_, peer_key.pub().key_to_bin()[10:])
-    assert channel.votes > 2.0
-    assert channel.votes < 2.1
+    assert 2.0 < channel.votes < 2.5
 
 
 @db_session

